lib/iris/util.py

```diff
@@ -1828,3 +1828,30 @@ def equalise_attributes(cubes):
         for key in list(cube.attributes.keys()):
             if key not in common_keys:
                 del cube.attributes[key]
+
+
+def _remove_ancils_and_cms_mapping_dims(cube, dims):
```

Comment:
After some consideration, I guess this is a reasonable place for it, as we already have other general-purpose private routines here.
I know it was my proposal, but can you think of a handier short name ?
(Anticipating complaint from @bjlittle [as on this one](https://github.com/SciTools/iris/pull/3562#discussion_r351185418) !)